console/console-init/ui/src/modules/device/components/DeviceListToolbar/DeviceListToolbar.test.tsx

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+import React from "react";
+import { render, fireEvent } from "@testing-library/react";
+import { MemoryRouter } from "react-router";
+import {
+  DeviceListToolbar,
+  IDeviceListToolbarProps
+} from "./DeviceListToolbar";
+import { DropdownItem } from "@patternfly/react-core";
+
+describe("<DeviceListToolbar />", () => {
+  const kebabItems: React.ReactNode[] = [
+    <DropdownItem onClick={jest.fn()}>Enable</DropdownItem>,
+    <DropdownItem onClick={jest.fn()}>Disable</DropdownItem>,
+    <DropdownItem onClick={jest.fn()}>Delete</DropdownItem>
+  ];
+
+  const bulkSelectItems: React.ReactNode[] = [
+    <DropdownItem key="item-1" onClick={jest.fn()}>
+      Select none (0 items)
+    </DropdownItem>,
+    <DropdownItem key="item-2" onClick={jest.fn()}>
+      Select page (10 items)
+    </DropdownItem>,
+    <DropdownItem key="item-3" onClick={jest.fn()}>
+      Select all (100 items)
+    </DropdownItem>
+  ];
+
+  const props: IDeviceListToolbarProps = {
+    itemCount: 100,
+    perPage: 10,
+    page: 1,
+    kebabItems: kebabItems,
+    onSetPage: jest.fn(),
+    onPerPageSelect: jest.fn(),
+    handleInputDeviceInfo: jest.fn(),
+    handleJSONUpload: jest.fn(),
+    isOpen: true,
+    handleOnSelect: jest.fn(),
+    handleOnToggle: jest.fn(),
+    isChecked: false,
+    items: bulkSelectItems,
+    handleOnChange: jest.fn()
+  };
+
+  it("should render the toolbar for device list", () => {
+    const { getByText } = render(
+      <MemoryRouter>
+        <DeviceListToolbar {...props} />
+      </MemoryRouter>
+    );
+
+    const createDeviceBtn = getByText("Add device");
+    const paginationNode = getByText("of 10");
+
+    expect(createDeviceBtn).toBeInTheDocument();
+    expect(paginationNode).toBeInTheDocument();
+  });
+
+  it("should show appropriate options in CreateDeviceButton", () => {
+    const { getByText } = render(
+      <MemoryRouter>
+        <DeviceListToolbar {...props} />
+      </MemoryRouter>
+    );
+
+    const createDeviceBtn = getByText("Add device");
+
+    fireEvent.click(createDeviceBtn);
+
+    const inputDeviceBtn = getByText("Input device info");
+    const uploadJSONBtn = getByText("Upload a JSON file");
+
+    expect(inputDeviceBtn).toBeInTheDocument();
+    expect(uploadJSONBtn).toBeInTheDocument();
+  });
+
+  it("should render appropriate options for BulkSelect", () => {
+    const { getByText } = render(
+      <MemoryRouter>
+        <DeviceListToolbar {...props} />
+      </MemoryRouter>
+    );
+
+    const dropDownItem0 = getByText("Select none (0 items)");
+    const dropDownItem1 = getByText("Select page (10 items)");
+    const dropDownItem2 = getByText("Select all (100 items)");
+
+    expect(dropDownItem0).toBeInTheDocument();
```

Comment:
expect is not required if you are using getByText. getByText internally manages this thing.

Reply:
Using expect to maintain consistency with the current codebase, also `toBeInTheDocument()` gives a better idea about purpose of the test.

Reply:
we should remove expect for all cases if we are using getByText. It is like doing same thing twice.